### PR TITLE
feat(skills): recognize .claude/skills and .agents/skills source roots

### PR DESCRIPTION
**Pull request by tomkis**
## Problem
Skill scan/import only recognized `skills/*` and top-level `*`. Repos that keep skills under `.claude/skills/` (Claude Code convention) reported no skills, so import silently failed. Ref #1637.

## Change
- Both scanners (api-server public-archive, agent-runtime clone) enumerate source roots in order `skills/` -> `.claude/skills/` -> `.agents/skills/` and union the results. Top-level `*` is consulted only as a fallback when no deliberate root matched.
- The union is deduped by skill name (frontmatter `name`, else basename, the install + `agent_skills` identity), first-found-wins. Dropped collisions are logged. This collapses a root-symlink pair (`.claude/skills` -> `.agents/skills`) to a single entry.
- Install resolver walks the same root order, so scan and install can never disagree on precedence.
- `SKILL_SOURCE_ROOTS` (ordered) and the pure `dedupeByName` helper live in `agent-runtime-api`, shared by both scanners and the resolver so the order cannot drift.
- Architecture doc updated with a Source Roots section.

## Out of scope
Recursive discovery, path-as-name, `.pi`/`.bob` roots, listLocal changes, "no skills found" messaging (#944).

### CODE REVIEW

File: packages/api-server/src/modules/skills/infrastructure/public-archive-scanner.ts [hunk 1]
```diff
@@ -116,27 +118,35 @@ export async function computeContentHash(absDir: string): Promise<string> {
...
-  for (const root of candidates) {
-    let entries: import("node:fs").Dirent[];
+  for (const root of SKILL_SOURCE_ROOTS) {
+    found.push(...(await skillDirsUnder(repoDir, path.join(repoDir, root))));
+  }
+  if (found.length > 0) return found;
+  return skillDirsUnder(repoDir, repoDir);
+}
+
+async function skillDirsUnder(
```
Comment (dam-code-guardian) [inline packages/api-server/src/modules/skills/infrastructure/public-archive-scanner.ts:128]: 🟢 **Suggestion:** This `skillDirsUnder` helper is now duplicated near-verbatim in `packages/agent-runtime/src/modules/skills/infrastructure/local-skill-repository.ts`. Since `SKILL_SOURCE_ROOTS` and `dedupeByName` already live in `agent-runtime-api`, hoisting this one-level-deep directory walk there too would eliminate the remaining chance of the two scanners' enumeration logic drifting apart.

File: packages/agent-runtime-api/src/modules/skills/source-roots.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  ".agents/skills",
+] as const;
+
+export interface DedupeByNameResult<T> {
+  kept: T[];
+  dropped: T[];
+}
+
+export function dedupeByName<T extends { name: string }>(
+  items: readonly T[],
```
Comment (dam-code-guardian) [inline packages/agent-runtime-api/src/modules/skills/source-roots.ts:13]: 🟢 **Suggestion:** `dedupeByName` is a pure function with clear edge cases (same-name collisions, root ordering, empty input). There's no existing test convention in these skills modules so this is optional, but it's a cheap, high-value unit-test target to lock in the first-found-wins contract.

**Review by dam-code-guardian — APPROVED** [review]
🛡️ **DAM** — ✅ Code Review @ `fdc86de`

## PR #1745: feat(skills): recognize .claude/skills and .agents/skills source roots
**Author:** tomkis | **Branch:** feat/skill-source-roots → main | **Changes:** +132 −51 (10 files)

### Summary
Teaches both skill scanners (api-server public-archive + agent-runtime clone) and the install resolver to enumerate skills across an ordered set of source roots (`skills/` → `.claude/skills/` → `.agents/skills/`), unioning results and deduping by skill name (first-found-wins), with top-level `*` kept only as a fallback. The ordering and the dedupe helper are centralized in `agent-runtime-api` so scan and install can never disagree on precedence.

### Findings
- ✅ **Looks good:** `SKILL_SOURCE_ROOTS` and the pure `dedupeByName` helper live in one shared package and are consumed by both scanners and the install resolver — precedence is single-sourced and cannot drift.
- ✅ **Looks good:** First-found-wins dedupe is preserved end-to-end: `findSkillDirsInClone` pushes in root order, and a stable `localeCompare` sort keeps equal-name entries in insertion (root) order, so dedupe still drops the later root deterministically. Dropped collisions are logged on every path.
- ✅ **Looks good:** The new required `log` dependency on `ComposeSkillsOptions` / `ScanDeps` is threaded through every caller (`server.ts` wires a stderr logger; `skills-service` forwards it to `runScan`).
- 🟢 **Suggestion:** `skillDirsUnder` is now duplicated near-verbatim across packages — `packages/agent-runtime/src/modules/skills/infrastructure/local-skill-repository.ts` and `packages/api-server/src/modules/skills/infrastructure/public-archive-scanner.ts`. Since `SKILL_SOURCE_ROOTS` and `dedupeByName` already live in `agent-runtime-api`, hoisting this one-level-deep directory walk there too would fully eliminate the chance of the two scanners' enumeration logic drifting apart. (`packages/api-server/src/modules/skills/infrastructure/public-archive-scanner.ts:128`)
- 🟢 **Suggestion:** `dedupeByName` is a pure function with clear edge cases (same-name collisions, ordering, empty input) and no co-located test. There's no existing test convention in these skills modules, so this is optional — but it's a cheap, high-value unit-test target if you want to lock in first-found-wins semantics. (`packages/agent-runtime-api/src/modules/skills/source-roots.ts`)

### Documentation Check (doc-drift)
✅ No documentation drift detected.

The skills subsystem page (`docs/architecture/skills.md`) was updated in the same PR: a new **Source Roots** section documents the union/dedupe/fallback semantics at architecture altitude (no code-shape leakage), `Last verified:` was bumped 2026-06-12 → 2026-06-23, and the `docs/ubiquitous-language.md` Skill Source entry was generalized. No ADR references, no volatile content leak.

### Verdict
APPROVE — Clean, well-factored refactor with single-sourced precedence, correct first-found-wins dedupe, complete dependency wiring, and architecture docs kept in sync; only minor optional suggestions.

---
_Review by [DAM](https://github.com/dam-agents/dam) · automated code guardian_
